### PR TITLE
Jeff/backup and restore

### PR DESCRIPTION
**Pull request by sofuture**
Want to merge this for the time being. Being replaced with a better solution in stolonctl, but this works in a simplistic case.

### CODE REVIEW

File: images/backup/backup.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+# create .pgpass
+
+echo "$STOLON_POSTGRES_SERVICE_HOST:$STOLON_POSTGRES_SERVICE_PORT:$STOLON_BACKUP_DB:stolon:$(cat /etc/secrets/stolon/password)" > ~/.pgpass
+chmod 0600 ~/.pgpass
+
+# pg_dump database
+
+filename=/backups/stolon-$STOLON_BACKUP_DB-$(date +%Y%m%dT%H%M%S).sql.gz
+pg_dump --host $STOLON_POSTGRES_SERVICE_HOST --port $STOLON_POSTGRES_SERVICE_PORT -U stolon -w $STOLON_BACKUP_DB | gzip > $filename
```
Comment (agalitsyn) [inline images/backup/backup.sh]: Also we can use http://www.pgbarman.org/ which is more advanced in settings

Comment (sofuture) [inline images/backup/backup.sh]: Interesting, that looks like a good option once this get's put into stolonctl.

File: resources/backup.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            value: "{{STOLON_BACKUP_DB}}"
+        volumeMounts:
+          - mountPath: /backups
+            name: backups
+          - mountPath: /etc/secrets/stolon
+            name: stolon
+      volumes:
+        - name: backups
+          hostPath:
+            path: /var/backups
```
Comment (agalitsyn) [inline resources/backup.yaml]: its ok for dev mode, but we should have a capability to change it to something replicated like s3. So maybe it's better not to save it on local FS and pass through mount to host, but send immediately after backup to some storage

File: resources/restore.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+          - mountPath: /etc/secrets/stolon
+            name: stolon
+      volumes:
+        - name: backups
+          hostPath:
+            path: /var/backups
+        - name: stolon
+          secret:
+            secretName: stolon
+      restartPolicy: OnFailure
```
Comment (agalitsyn) [inline resources/restore.yaml]: I guess if we backup something, we maybe want that it restores, so its better to periodically check that backups can be restored on some dummy DB. From my experience it not always works, sometime backups are broken and when it needed (backup always a last hope) it not work

File: images/backup/backup.sh [hunk 4]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+set -e
+
+# create .pgpass
+
+echo "$STOLON_POSTGRES_SERVICE_HOST:$STOLON_POSTGRES_SERVICE_PORT:$STOLON_BACKUP_DB:stolon:$(cat /etc/secrets/stolon/password)" > ~/.pgpass
```
Comment (agalitsyn) [inline images/backup/backup.sh]: I cant find where this vars were defined

File: resources/backup.yaml (hunk 2 above)
Comment (klizhentas) [inline resources/backup.yaml]: the plan is to use s3 api and push to pithos.io

**Comment by agalitsyn** [comment]
ok, comments are for future fixes, now :shipit:

File: images/backup/backup.sh (hunk 4 above)
Comment (sofuture) [inline images/backup/backup.sh]: answered out of band: they come from k8s services in environment automagically

**Pull request merged**